dashboard/app/assets/javascripts/header.js.erb

```diff
@@ -0,0 +1,106 @@
+/**
+ * Dynamic header generation and event bindings for header actions.
+ */
+
+if (!window.dashboard) {
+  window.dashboard = {};
+}
+
+/**
+ * @param scriptData{{
+ *   finishLink: string,
+ *   statsPath: string,
+ *   showStageLinks: boolean,
+ *   trophies: Object,
+ *   currentLevelIndex: number,
+ *   scriptId: number,
+ *   scriptLevelId: number,
+ *   levels: Array.<{
```

Comment:
GitHub highlights this on the diff but it's inside a comment and correct per https://developers.google.com/closure/compiler/docs/js-for-compiler#types